app/controllers/parent_objects_controller.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+class ParentObjectsController < ApplicationController
+  before_action :set_parent_object, only: [:show, :edit, :update, :destroy]
+
+  # GET /parent_objects
+  # GET /parent_objects.json
+  def index
+    @parent_objects = ParentObject.all
+  end
+
+  # GET /parent_objects/1
+  # GET /parent_objects/1.json
+  def show; end
+
+  # GET /parent_objects/new
+  def new
+    @parent_object = ParentObject.new
+  end
+
+  # GET /parent_objects/1/edit
+  def edit; end
+
+  def build_metadata_cloud_url(metadata_source, lb_record = "")
+    case metadata_source
+    when "ladybird"
+      "https://metadata-api-test.library.yale.edu/metadatacloud/api/ladybird/oid/#{@parent_object.oid}"
+    when "ils"
+      identifier_block = if lb_record["orbisBarcode"].nil?
+                           "/bib/#{lb_record['orbisRecord']}"
+                         else
+                           "/barcode/#{lb_record['orbisBarcode']}?bib=#{lb_record['orbisRecord']}"
+                         end
+      "https://metadata-api-test.library.yale.edu/metadatacloud/api/ils#{identifier_block}"
+    when "aspace"
+      "https://metadata-api-test.library.yale.edu/metadatacloud/api/aspace#{lb_record['archiveSpaceUri']}"
+    end
+  end
+
+  def mc_get(mc_url)
+    metadata_cloud_username = ENV["MC_USER"]
+    metadata_cloud_password = ENV["MC_PW"]
+    HTTP.basic_auth(user: metadata_cloud_username, pass: metadata_cloud_password).get(mc_url)
+  end
+
+  # Takes a JSON record from the MetadataCloud and saves the Ladybird-specific info to the DB
+  def save_ladybird_info_to_db(lb_record)
```

Comment:
this should be inverted to be @parent_object.lb_record = lb_record. all this logic can then live in lb_record=(value)